ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -783,15 +783,9 @@ private void doWorkInternal()
             // we carry on scanning even if scanning is SUSPENDED because the suspending thread
             // might be waiting on us to complete and we don't want to risk deadlocking it by waiting
             // here for a resume.
-            // if we have been TERMINATED we bail out now
+            // if we have been TERMINATED we still run phase 2 since some modules (XARecoveryModule for example)
+            // require phase 2 if phase 1 has ran
             // n.b. if we give up here the caller is responsible for clearing the active scan
```

Comment:
i think this line could be removed - did you check the history to know about whether there was a need to bail out?